### ISSUE TITLE
Report summary stats of animal population file used for herd initialization

### ISSUE THREAD

**Issue by KFosterReed** [RuminantFarmSystems/RuFaS#2065]
**Is your feature request related to a problem? Please describe.**
It is helpful to know the characteristics of the animal population that is used to initialize the herd but we don't currently report any of this information to the output manager

**Describe the solution you'd like**
I would like to add to the summary stats that are created by the `get_herd_summary()` function in `animal_population.py` to include some more useful stats like average bodyweight for each animal class, average milk production, and the number of cows in each parity. Before resolving this issue we should poll other SMEs to see what additional summary stats would be useful

**Describe alternatives you've considered**
we could create a standalone task to summarize the animal population file

**Additional context**

**Target Deadline**
Soon after animal module refresh is complete

**Comment by matthew7838** [comment]
This issue is blocked by #2041.

**Comment by allisterakun** [comment]
@KFosterReed please advise the desired herd summary statistics.

**Comment by KFosterReed** [comment]
Initial thoughts:

- average BW for each animal type
- distribution of ages for each animal type
- number of dry and lactating cows
- number of cows in each parity

@jadamchick @krb78 @JoeWaddell any others you can think of?

**Comment by KFosterReed** [comment]
@tomhuhh also if you have any requests to add, please do!

**Comment by allisterakun** [comment]
Just confirming that we are interested in the summary statistics for the final animal population used for the simulation. Is that correct?

**Comment by tomhuhh** [comment]
> Just confirming that we are interested in the summary statistics for the final animal population used for the simulation. Is that correct?

Correct, that's my understanding as well. Regarding the statistics outputs for the initialized animal population, I would consider extending the request to include counts for all animal types (calves, growing heifers, close-up heifers, lactating cows, and dry cows) instead of just dry and lactating cows.

**Comment by allisterakun** [comment]
Also, what does "distribution of ages for each animal type" mean? Is it simply the average age for each animal type?

**Comment by JoeWaddell** [comment]
>what does "distribution of ages for each animal type" mean?

There's interest in knowing whether the ages approximate a uniform distribution, versus a normal distribution, for instance. That's going to be a little more challenging to summarize in a digestible way for most, but once I've pushed my local commits from #2252 I can help explore how we can institute this kind of check.

**Comment by allisterakun** [comment]
I have just put together a PR [[Animal] Report Herd Summary Statistics for Starting Animal Population #2314](https://github.com/RuminantFarmSystems/MASM/pull/2314) for all requested stats, but the age distribution. 

Please take an initial look and verify if it works as expected. 



> That's going to be a little more challenging to summarize in a digestible way for most

Regarding the calculation and reporting of the age distribution, I have no idea how to summarize and report a distribution at the moment, hoping to get more help on this from @JoeWaddell.

**Comment by krb78** [comment]
I think this list looks great. Silly question: is it worth knowing anything about milk production or because that is an input we shouldn't bother looking for it to be an output? 

The only thing I might consider adding would be a % lactating herd DNB summary statistic output. (@allisterakun I assume this is doable because I know you code animals by repro status, but if it's not, don't stress about it)

**Comment by allisterakun** [comment]
> a % lactating herd DNB summary statistic output

Regarding the % DNB output, I want to confirm that to calculate this, we find the number of lactating cows that has been marked as do_not_breed, and divide that by the total number of lactating cows. Is that correct?

On second thought, for the saved animals, I don't think we are saving the `do_not_breed` flag. So when loading them back into the simulation, we initialize all animals as False for `do_not_breed`. Currently, this flag will get updated every time the animal goes through a round of reproduction updates. Therefore, if we calculate this output, we will always have 0% of the lactating cows marked as DNB.

**Comment by jadamchick** [comment]
Hi all! I agree it looks like a good list. Two minor comments- 
* agree that average milk per cow per day would be a nice-to-have
* I know we aren't really distinguishing between breeds right now, but eventually breed might be a nice output as well. Even as just a reminder if you are simulating a herd with one breed but starting with an animal population of a different breed (which might even be a good thing to flag with a warning or something, depending on the impact of breed)

Thanks!
